### PR TITLE
Replace fmt.Errorf with errors.New when there are no values to format…

### PR DESCRIPTION
**Pull request by natemurthy**
The `fmt.Errorf` function is just a convenience wrapper around `errors.New`. If you just want to create an error string with no interpolation why not just use the latter?

### CODE REVIEW

**Comment by neolit123** [comment]
the `fmt.Errorf` -> `errors.New` change is not bad it has performance implications.
but the PR touches other areas especially in kubeadm, please rebase your commits against the latest master.

**Comment by natemurthy** [comment]
@neolit123 Rebased. Any suggestions as to what I should put for release note?

Btw, this is approximately how many other `fmt.Errorf` expressions I could replace with `errors.New`:
```
nmurthy:kubernetes% rg "return fmt.Errorf\(\"(.*?)\"\)" | wc -l
    1692
```

**Comment by neolit123** [comment]
> @neolit123 Rebased. 

thanks but it seems that it still touches other code?

> Any suggestions as to what I should put for release note?

given this isn't user facing there is no need for a release note.

> nmurthy:kubernetes% rg "return fmt.Errorf\(\"(.*?)\"\)" | wc -l
    1692

these are bug numbers, but the problem is that if you submit it all in as a single PR it might take _a while_ before it's merged, unless a top level approver comes by.

File: cmd/kubeadm/app/cmd/upgrade/node.go [hunk 1]
```diff
@@ -108,59 +95,10 @@ func NewCmdUpgradeNodeConfig() *cobra.Command {
...
-	}
-
-	options.AddKubeConfigFlag(cmd.Flags(), &flags.kubeConfigPath)
-	cmd.Flags().BoolVar(&flags.dryRun, "dry-run", flags.dryRun, "Do not change any state, just output the actions that would be performed.")
-
-	//TODO: following values should retrieved form the kubeadm-config config map; remove as soon as the new config wil be in place
-	cmd.Flags().StringVar(&flags.advertiseAddress, "apiserver-advertise-address", flags.advertiseAddress, "If the node is joining as a master, the IP address the API Server will advertise it's listening on.")
-	cmd.Flags().StringVar(&flags.nodeName, "node-name", flags.nodeName, "Specify the node name.")
-	return cmd
-}
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/upgrade/node.go]: [1]

File: cmd/kubeadm/app/cmd/upgrade/node.go [hunk 2]
```diff
@@ -219,43 +157,3 @@ func printFilesIfDryRunning(dryRun bool, kubeletDir string) error {
...
-		return DryRunStaticPodUpgrade(cfg)
-	}
-
-	if err := PerformStaticPodUpgrade(client, waiter, cfg, false); err != nil {
-		return fmt.Errorf("Couldn't complete the static pod upgrade: %v", err)
-	}
-
-	fmt.Println("[upgrade] The control plane instance for this node was successfully updated!")
-	return nil
-}
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/upgrade/node.go]: ^ [1]

File: cmd/kubeadm/app/phases/upgrade/postupgrade.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 	}
 
 	// Rotate the kube-apiserver cert and key if needed
-	if err := BackupAPIServerCertIfNeeded(cfg, dryRun); err != nil {
+	if err := backupAPIServerCertIfNeeded(cfg, dryRun); err != nil {
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/upgrade/postupgrade.go]: [1]

File: cmd/kubeadm/app/phases/upgrade/postupgrade.go [hunk 4]
```diff
@@ -172,8 +172,7 @@ func upgradeToSelfHosting(client clientset.Interface, cfg *kubeadmapi.InitConfig
 	return nil
 }
 
-// BackupAPIServerCertIfNeeded rotates the kube-apiserver certificate if older than 180 days
-func BackupAPIServerCertIfNeeded(cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
+func backupAPIServerCertIfNeeded(cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/upgrade/postupgrade.go]: [1]

File: cmd/kubeadm/app/phases/upgrade/prepull.go [hunk 5]
```diff
@@ -159,11 +160,6 @@ func buildPrePullDaemonSet(component, image string) *apps.DaemonSet {
 			Namespace: metav1.NamespaceSystem,
 		},
 		Spec: apps.DaemonSetSpec{
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"k8s-app": addPrepullPrefix(component),
-				},
-			},
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/upgrade/prepull.go]: [1]

File: cmd/kubeadm/app/phases/upgrade/staticpods.go [hunk 6]
```diff
@@ -185,29 +185,28 @@ func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticP
...
+				return fmt.Errorf("failed to upgrade the %s healthcheck certificate and key: %v", constants.Etcd, err)
 			}
-			if component == constants.KubeAPIServer {
-				if err := certsphase.KubeadmCertEtcdAPIClient.CreateFromCA(cfg, caCert, caKey); err != nil {
-					return fmt.Errorf("failed to upgrade the %s %s-client certificate and key: %v", constants.KubeAPIServer, constants.Etcd, err)
-				}
+		}
+		if component == constants.KubeAPIServer {
+			if err := certsphase.KubeadmCertEtcdAPIClient.CreateFromCA(cfg, caCert, caKey); err != nil {
+				return fmt.Errorf("failed to upgrade the %s %s-client certificate and key: %v", constants.KubeAPIServer, constants.Etcd, err)
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/upgrade/staticpods.go]: [1]

File: cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go [hunk 7]
```diff
@@ -31,10 +31,10 @@ import (
 )
 
 const (
-	// NodesKubeadmConfigClusterRoleName sets the name for the ClusterRole that allows
+	// BootstrapDiscoveryClusterRoleName sets the name for the ClusterRole that allows
 	// the bootstrap tokens to access the kubeadm-config ConfigMap during the node bootstrap/discovery
-	// or during upgrade nodes
-	NodesKubeadmConfigClusterRoleName = "kubeadm:nodes-kubeadm-config"
+	// phase for additional master nodes
+	BootstrapDiscoveryClusterRoleName = "kubeadm:bootstrap-discovery-kubeadm-config"
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go]: [1]

File: cmd/kubeadm/app/cmd/upgrade/node.go [hunk 8]
```diff
@@ -63,13 +58,6 @@ type nodeUpgradeFlags struct {
 	dryRun            bool
 }
 
-type controlplaneUpgradeFlags struct {
-	kubeConfigPath   string
-	advertiseAddress string
-	nodeName         string
-	dryRun           bool
-}
-
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/upgrade/node.go]: [1]

**Review by neolit123 — CHANGES_REQUESTED** [review]
/hold
to clarify - please remove code changes unrelated to the PR (marked with [1]). thanks!

**Comment by natemurthy** [comment]
I recently ran a
```
git fetch upstream
git rebase upstream/master
git push -f origin master
```
I don't understand where all these other code changes came from.

**Comment by natemurthy** [comment]
This isn't worth the effort to me any more to constantly keep my fork up-to-date with master upstream.

**Pull request closed without merging**